agent/tests/test_user_instrument.c

```diff
@@ -52,14 +52,124 @@ static void test_op_array_wraprec(TSRMLS_D) {
   tlib_php_request_end();
 }
 
+static void test_get_wraprec_by_name() {
+#if ZEND_MODULE_API_NO < ZEND_7_3_X_API_NO
+  return;
+#else
+  zend_op_array oparray = {.function_name = (void*)1};
+  zend_function zend_func = {0};
+  zend_string* scope_name = NULL;
+  zend_class_entry ce = {0};
+  nruserfn_t* wraprec = NULL;
+  char* name_str = "my_func_name";
+
+  tlib_php_request_start();
+
+  /*
+   * NULL if there's no wraprecs in the internal list.
+   */
+  tlib_pass_if_ptr_equal("obtain instrumented function",
+                         nr_php_get_wraprec_by_name(&zend_func), NULL);
+
+  wraprec = nr_php_add_custom_tracer_named(
+      "ClassNoMatch::functionNoMatch",
+      nr_strlen("ClassNoMatch::functionNoMatch"));
+  wraprec = nr_php_add_custom_tracer_named("functionNoMatch2",
+                                           nr_strlen("functionNoMatch2"));
+  wraprec = nr_php_add_custom_tracer_named(name_str, nr_strlen(name_str));
+
+  /*
+   * NULL if zend_function is NULL.
+   */
+  tlib_pass_if_ptr_equal("obtain instrumented function",
+                         nr_php_get_wraprec_by_name(NULL), NULL);
+
+  /*
+   * NULL if function name is NULL.
+   */
+  tlib_pass_if_ptr_equal("obtain instrumented function",
+                         nr_php_get_wraprec_by_name(&zend_func), NULL);
+
+  /*
+   * NULL if name is correct but type is wrong.
+   */
+  zend_func.common.function_name
```

Comment:
Maybe set type explicitly for zend_func (like below when set to user function) for clarity and to improve test robustness?

Reply:
See commit [2b4a42a](https://github.com/newrelic/newrelic-php-agent/pull/541/commits/2b4a42aaa23d1c19063cdbeae013c62131518337)